src/connector.cpp

```diff
@@ -3517,19 +3437,22 @@ void HostConnector::hostEnsureStereoChain(const uint8_t row, const uint8_t block
 
 // --------------------------------------------------------------------------------------------------------------------
 
-void HostConnector::hostSetupSideIO(const uint8_t row,
+void HostConnector::hostSetupSideIO(const uint8_t preset,
+                                    const uint8_t row,
                                     const uint8_t block,
                                     const HostBlockPair hbp,
                                     const Lv2Plugin* plugin)
 {
-    mod_log_debug("hostSetupSideIO(%u, %u, {%u, %u}, %p)", row, block, hbp.id, hbp.pair, plugin);
+    mod_log_debug("hostSetupSideIO(%u, %u, %u, {%u, %u}, %p)", preset, row, block, hbp.id, hbp.pair, plugin);
     assert(row < NUM_BLOCK_CHAIN_ROWS);
     assert(block < NUM_BLOCKS_PER_PRESET);
     assert(hbp.id != kMaxHostInstances);
 
-    const Block& blockdata(_current.chains[row].blocks[block]);
+    const bool active = _current.preset == preset;
+    const Block& blockdata(active ? _current.chains[row].blocks[block] : _presets[preset].chains[row].blocks[block]);
     assert(!isNullBlock(blockdata));
-
+    ChainRow& nextChainRow = active ? _current.chains[row + 1] : _presets[preset].chains[row + 1];
```

Comment:
this definition happens to early, must be moved after the "row + 1 < NUM_BLOCK_CHAIN_ROWS" check below